examples/with-evm-web3-react-next/chains.ts

```diff
@@ -0,0 +1,164 @@
+  name: 'Ether',
+  symbol: 'ETH',
+  decimals: 18,
+};
+
+const MATIC: AddEthereumChainParameter['nativeCurrency'] = {
+  name: 'Matic',
+  symbol: 'MATIC',
+  decimals: 18,
+};
+
+const CELO: AddEthereumChainParameter['nativeCurrency'] = {
+  name: 'Celo',
+  symbol: 'CELO',
+  decimals: 18,
+};
+
+interface BasicChainInformation {
+  urls: string[];
+  name: string;
+}
+
+interface ExtendedChainInformation extends BasicChainInformation {
+  nativeCurrency: AddEthereumChainParameter['nativeCurrency'];
+  blockExplorerUrls: AddEthereumChainParameter['blockExplorerUrls'];
+}
+
+function isExtendedChainInformation(
+  chainInformation: BasicChainInformation | ExtendedChainInformation
+): chainInformation is ExtendedChainInformation {
+  return !!(chainInformation as ExtendedChainInformation).nativeCurrency;
+}
+
+export function getAddChainParameters(
+  chainId: number
+): AddEthereumChainParameter | number {
+  const chainInformation = CHAINS[chainId];
+  if (isExtendedChainInformation(chainInformation)) {
+    return {
+      chainId,
+      chainName: chainInformation.name,
+      nativeCurrency: chainInformation.nativeCurrency,
+      rpcUrls: chainInformation.urls,
+      blockExplorerUrls: chainInformation.blockExplorerUrls,
+    };
+  } else {
+    return chainId;
+  }
+}
+
+const getInfuraUrlFor = (network: string) =>
+  process.env.infuraKey
+    ? `https://${network}.infura.io/v3/${process.env.infuraKey}`
+    : undefined;
+const getAlchemyUrlFor = (network: string) =>
+  process.env.alchemyKey
+    ? `https://${network}.alchemyapi.io/v2/${process.env.alchemyKey}`
+    : undefined;
+
+type ChainConfig = {
+  [chainId: number]: BasicChainInformation | ExtendedChainInformation;
+};
+
+export const MAINNET_CHAINS: ChainConfig = {
+  1: {
+    urls: [
+      getInfuraUrlFor('mainnet'),
+      getAlchemyUrlFor('eth-mainnet'),
+      'https://cloudflare-eth.com',
+    ].filter(Boolean),
+    name: 'Mainnet',
+  },
+  10: {
+    urls: [
+      getInfuraUrlFor('optimism-mainnet'),
+      'https://mainnet.optimism.io',
+    ].filter(Boolean),
+    name: 'Optimism',
+    nativeCurrency: ETH,
+    blockExplorerUrls: ['https://optimistic.etherscan.io'],
+  },
+  42161: {
+    urls: [
+      getInfuraUrlFor('arbitrum-mainnet'),
+      'https://arb1.arbitrum.io/rpc',
+    ].filter(Boolean),
+    name: 'Arbitrum One',
+    nativeCurrency: ETH,
+    blockExplorerUrls: ['https://arbiscan.io'],
+  },
+  137: {
+    urls: [
+      getInfuraUrlFor('polygon-mainnet'),
+      'https://polygon-rpc.com',
+    ].filter(Boolean),
+    name: 'Polygon Mainnet',
+    nativeCurrency: MATIC,
+    blockExplorerUrls: ['https://polygonscan.com'],
+  },
+  42220: {
+    urls: ['https://forno.celo.org'],
+    name: 'Celo',
+    nativeCurrency: CELO,
+    blockExplorerUrls: ['https://explorer.celo.org'],
+  },
+};
+
+export const TESTNET_CHAINS: ChainConfig = {
+  5: {
+    urls: [getInfuraUrlFor('goerli')].filter(Boolean),
```

Comment:
Guess we can pass a public rpc url for `goerli` too so that it doesn't look like the demo is broken

Reply:
Will be okay since we make rpcUrl optional and add default rpc 0ec1918560f290a5983e2446a3bfac3e9b1c695f